x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/history/SnapshotLifecycleTemplateRegistry.java

```diff
@@ -37,8 +37,9 @@
     // version 1: initial
     // version 2: converted to hidden index
     // version 3: templates moved to composable templates
-    // version 4: add `allow_auto_create` setting
-    public static final int INDEX_TEMPLATE_VERSION = 4;
+    // version 4:converted data stream
+    // version 5: add `allow_auto_create` setting
```

Comment:
According to the [`AutoCreateAction` class](https://github.com/elastic/elasticsearch/blob/41b2fc6588c839ff91af7e354eed44726295f523/server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java#L120), data streams are always auto-created. I'm not sure whether that answers your question?

Reply:
This is an interesting point. Right now, there's no benefit to adding `allow_auto_create` to data stream templates. However, it seems to me that the user experience would be more consistent if the auto-creation of data streams was also subject to the `auto_create_index` setting. Then an administrator could stop their users from e.g. creating a slew of `logs-*-*` data streams.

@martijnvg I believe you added the code to auto-create data streams - what is your view here?

Reply:
I've changed `AutoCreateAction` so that data streams are now also subject to `auto_create_index`, and added an integration test. I tried reworking the exist auto-create IT, but as it's under `qa` the data stream API wasn't available.